services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/application/ApplicationDataMapper.kt

```diff
@@ -2,57 +2,64 @@ package fi.hel.haitaton.hanke.application
 
 import fi.hel.haitaton.hanke.COORDINATE_SYSTEM_URN
 import fi.hel.haitaton.hanke.allu.AlluCableReportApplicationData
+import fi.hel.haitaton.hanke.application.AlluDataError.EMPTY_OR_NULL
+import fi.hel.haitaton.hanke.application.AlluDataError.NULL
 import fi.hel.haitaton.hanke.geometria.UnsupportedCoordinateSystemException
+import java.time.ZonedDateTime
 import org.geojson.GeometryCollection
 import org.geojson.Polygon
 
 object ApplicationDataMapper {
 
-    fun toAlluData(applicationData: CableReportApplicationData): AlluCableReportApplicationData =
-        AlluCableReportApplicationData(
-            applicationData.name,
-            applicationData.customerWithContacts.toAlluData("applicationData.customerWithContacts"),
-            getGeometry(applicationData),
-            applicationData.startTime
-                ?: throw AlluDataException("applicationData.startTime", AlluDataError.NULL),
-            applicationData.endTime
-                ?: throw AlluDataException("applicationData.endTime", AlluDataError.NULL),
-            applicationData.pendingOnClient,
-            applicationData.identificationNumber,
-            applicationData.clientApplicationKind,
-            getWorkDescription(applicationData),
-            applicationData.contractorWithContacts.toAlluData(
-                "applicationData.contractorWithContacts"
-            ),
-            applicationData.postalAddress?.toAlluData(),
-            applicationData.representativeWithContacts?.toAlluData(
-                "applicationData.representativeWithContacts"
-            ),
-            applicationData.invoicingCustomer?.toAlluData("applicationData.invoicingCustomer"),
-            applicationData.customerReference,
-            applicationData.area,
-            applicationData.propertyDeveloperWithContacts?.toAlluData(
-                "applicationData.propertyDeveloperWithContacts"
-            ),
-            applicationData.constructionWork,
-            applicationData.maintenanceWork,
-            applicationData.emergencyWork,
-            applicationData.propertyConnectivity
-        )
+    private const val BASE = "applicationData."
 
-    private fun getWorkDescription(applicationData: CableReportApplicationData): String {
-        val rockExcavation =
-            applicationData.rockExcavation
-                ?: throw AlluDataException("applicationData.rockExcavation", AlluDataError.NULL)
-        return applicationData.workDescription +
-            if (rockExcavation) "\nLouhitaan" else "\nEi louhita"
-    }
+    private const val AREAS = "areas"
+    private const val CONTRACTOR = "contractorWithContacts"
+    private const val CUSTOMER = "customerWithContacts"
+    private const val DEVELOPER = "propertyDeveloperWithContacts"
+    private const val EXCAVATION = "rockExcavation"
+    private const val INVOICING = "invoicingCustomer"
```

Comment:
I don't see much point in defining a constant for a value that's used only once. Especially here, where the value should be identical to the field name. I think using a constant that has a different name from the field in question adds an unneeded layer of obfuscation.

Reply:
I like using constants to lessen the likelihood of typos when repeating the same string values. Currently they are not repeating very much so I guess they can be removed.

Reply:
These are now removed.